File: solutions/c/prime-factors/1/prime_factors.c
#include "prime_factors.h"

size_t find_factors(uint64_t n, uint64_t factors[static MAXFACTORS]) {
    if (n <= 0) {
        return -1;
    }
    size_t size = 0;
    uint64_t p = 2;
    while (n != 1) {
        while (n % p == 0) {
            factors[size] = p;
            n /= p;
            size += 1;
        }
        p += 1;
    }
    return size;
}

File: solutions/javascript/prime-factors/1/prime-factors.js
//
// This is only a SKELETON file for the 'Prime Factors' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const primeFactors = (n) => {
  let p = 2;
  let ps = [];
  while (n != 1) {
    while (n % p !== 0) {
      p++;
      if (p * p > n) {
        p = n;
      }
    }
    while (n % p === 0) {
      ps.push(p);
      n /= p;
    }
  }
  return ps;
};
